src/mscorlib/src/System/String.cs

```diff
@@ -411,7 +415,61 @@ private unsafe static bool EqualsHelper(String strA, String strB)
                 return (length <= 0);
             }
         }
-        
+
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        [ReliabilityContract(Consistency.WillNotCorruptState, Cer.MayFail)]
+        private unsafe static bool OrdinalCompareSubstring(String strA, String strB, int length)
+        {
+            Contract.Requires(strA != null);
+            Contract.Requires(strB != null);
+            Contract.Requires(strA.Length <= length);
+            Contract.Requires(strB.Length <= length);
+
+            fixed (char* ap = &strA.m_firstChar) fixed (char* bp = &strB.m_firstChar)
+            {
+                char* a = ap;
+                char* b = bp;
+
+#if AMD64
+#if ALIGN_ACCESS
+    #error Needs porting for ALIGN_ACCESS (https://github.com/dotnet/corert/issues/430) 
+#endif // ALIGN_ACCESS
+
+                while (length >= 12)
+                {
+                    if (*(long*)a != *(long*)b) goto ReturnFalse;
+                    if (*(long*)(a + 4) != *(long*)(b + 4)) goto ReturnFalse;
+                    if (*(long*)(a + 8) != *(long*)(b + 8)) goto ReturnFalse;
+                    a += 12; b += 12; length -= 12;
```

Comment:
I'm a bit torn on that one. It is faster for longer equal strings but it is a tax on early exits (differs within the first 12 characters) which should be more common.

Long = 252 char equal strings
Short = 252 char strings that differ at index 1

                Method |    AvrTime |    StdDev |           op/s |
---------------------- |-----------: |----------: |---------------: |
  LongStartsWithIncAfter | 33.7967 ns | 1.9410 ns | 29,678,807.37 |
 LongStartsWithIncBefore | 33.2239 ns | 1.8214 ns | 30,183,392.10 |
  ShortStartsWithIncAfter | 7.0180 ns | 0.5621 ns | 142,983,128.85 |
 ShortStartsWithIncBefore | 7.2873 ns | 0.5598 ns | 137,668,403.59 |